### PR TITLE
Apis 1801

### PR DESCRIPTION
**Pull request by peteslater-ee**
Adds the ability to create a test organisation with a LISA service.

See [APIS-1801](https://jira.tools.tax.service.gov.uk/browse/APIS-1801) for more information.

### CODE REVIEW

File: app/uk/gov/hmrc/testuser/models/TestUser.scala [hunk 1]
```diff
@@ -170,4 +173,16 @@ object MtdItId extends Modulus23Check with (String => MtdItId) {
...
+case class LisaManagerReferenceNumber(lisaManagerReferenceNumber: String) extends TaxIdentifier with SimpleName {
+  override def toString = lisaManagerReferenceNumber
+  val name = "lisaManagerReferenceNumber"
+  def value = lisaManagerReferenceNumber
+}
+
+object LisaManagerReferenceNumber extends (String => LisaManagerReferenceNumber) {
+  implicit val lisaManRefNumWrite: Writes[LisaManagerReferenceNumber] = new SimpleObjectWrites[LisaManagerReferenceNumber](_.value)
+  implicit val lisaManRefNumRead: Reads[LisaManagerReferenceNumber] = new SimpleObjectReads[LisaManagerReferenceNumber]("lisaManagerReferenceNumber", LisaManagerReferenceNumber.apply)
+}
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/testuser/models/TestUser.scala]: Newline.

File: app/uk/gov/hmrc/testuser/services/Generator.scala [hunk 2]
```diff
@@ -93,4 +96,13 @@ class MtdItIdGenerator(random: Random = new Random) extends Modulus23Check {
...
+}
+
+class LisaGenerator(random: Random = new Random) extends Modulus23Check {
+  def this(seed: Int) = this(new scala.util.Random(seed))
+
+  def nextLisaManRefNum: LisaManagerReferenceNumber = {
+    val randomCode = f"${random.nextInt(999999)}%06d"
+    LisaManagerReferenceNumber(s"Z$randomCode")
+  }
 }
```
Comment (AndySpaven) [inline app/uk/gov/hmrc/testuser/services/Generator.scala]: Newline

File: test/unit/uk/gov/hmrc/testuser/controllers/AuthenticationControllerSpec.scala [hunk 3]
```diff
@@ -111,4 +112,4 @@ class AuthenticationControllerSpec extends UnitSpec with MockitoSugar with WithF
       }
     }
   }
-}
+}
```
Comment (AndySpaven) [inline test/unit/uk/gov/hmrc/testuser/controllers/AuthenticationControllerSpec.scala]: \n

File: test/unit/uk/gov/hmrc/testuser/services/GeneratorSpec.scala [hunk 4]
```diff
@@ -188,4 +199,4 @@ object CustomMatchers {
   }
 
   def haveDifferentPropertiesThan(right: TestUser) = new HaveDifferentPropertiesThan(right)
-}
+}
```
Comment (AndySpaven) [inline test/unit/uk/gov/hmrc/testuser/services/GeneratorSpec.scala]: \n

File: test/unit/uk/gov/hmrc/testuser/controllers/TestUserControllerSpec.scala [hunk 5]
```diff
@@ -163,4 +163,4 @@ class TestUserControllerSpec extends UnitSpec with MockitoSugar with WithFakeApp
       }
     }
   }
-}
+}
```
Comment (AndySpaven) [inline test/unit/uk/gov/hmrc/testuser/controllers/TestUserControllerSpec.scala]: \n

**Review by AndySpaven — APPROVED** [review]
LGTM

File: app/uk/gov/hmrc/testuser/services/Generator.scala [hunk 6]
```diff
@@ -19,7 +19,7 @@ package uk.gov.hmrc.testuser.services
 import org.scalacheck.Gen
 import uk.gov.hmrc.domain._
 import uk.gov.hmrc.testuser.models.ServiceName._
-import uk.gov.hmrc.testuser.models.{MtdItId, TestAgent, TestIndividual, TestOrganisation}
+import uk.gov.hmrc.testuser.models.{ServiceName => _, _}
```
Comment (Tobias1087) [inline app/uk/gov/hmrc/testuser/services/Generator.scala:22]: Bit of a weird import

File: test/unit/uk/gov/hmrc/testuser/connectors/GovernmentGatewayLoginSpec.scala [hunk 7]
```diff
@@ -83,24 +85,24 @@ class GovernmentGatewayLoginSpec extends UnitSpec {
...
+      val organisation = TestOrganisation(user, password, Some(saUtr), Some(nino), Some(mtdItId), Some(empRef), Some(ctUtr), Some(vrn), Some(lisaManRefNum),
+        Seq(AGENT_SERVICES, NATIONAL_INSURANCE, CORPORATION_TAX, SUBMIT_VAT_RETURNS, LISA))
 
-        GovernmentGatewayLogin(organisation).enrolments should contain theSameElementsAs Seq(ctEnrolment, vatEnrolment)
+      GovernmentGatewayLogin(organisation).enrolments should contain theSameElementsAs Seq(ctEnrolment, vatEnrolment, lisaEnrolment)
     }
   }
 
-}
+}
```
Comment (Tobias1087) [inline test/unit/uk/gov/hmrc/testuser/connectors/GovernmentGatewayLoginSpec.scala]: /n

File: app/uk/gov/hmrc/testuser/services/Generator.scala (hunk 6 above)
Comment (peteslater-ee) [reply app/uk/gov/hmrc/testuser/services/Generator.scala:22]: It looks like it is a thing - http://docs.scala-lang.org/cheatsheets/
It imports everything from the package except the things that are filtered.

**Pull request merged**